File: Projet Angular2/my-app/src/app/model/categorie.ts
//sera transdere au format JSON
import { Produit } from "app/model/produit";

export class Categorie {

    idCategorie : number;
    nomCategorie : string;
    description : string;
    nomPhoto : string;
    produits : Produit[];

    constructor (idCategorie : number = 0, 
             nomCategorie : string = "?",
             description : string = "?",
             nomPhoto : string = "?",
             produits : Produit[] = []
             ) {
    
        this.idCategorie = idCategorie;
        this.nomCategorie = nomCategorie;
        this.description = description;
        this.nomPhoto = nomPhoto;
        this.produits = produits;
   
    }


}

File: Projet JavaEE Spring WebServiceRest/BMFTablet/src/main/java/com/tablet/bmf/entities/Categorie.java
package com.tablet.bmf.entities;

import java.io.Serializable;
import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.validation.constraints.Size;

import org.hibernate.validator.constraints.NotEmpty;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
public class Categorie implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long idCategorie;

	@NotEmpty
	@Size(max = 30)
	private String nomCategorie;

	private String description;

	private String nomPhoto;

	@OneToMany(mappedBy = "categorie")
	@JsonIgnore
	private List<Produit> produits;

	public Long getIdCategorie() {
		return idCategorie;
	}

	public void setIdCategorie(Long idcategorie) {
		this.idCategorie = idcategorie;
	}

	public String getNomCategorie() {
		return nomCategorie;
	}

	public void setNomCategorie(String nomCategorie) {
		this.nomCategorie = nomCategorie;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getNomPhoto() {
		return nomPhoto;
	}

	public void setNomPhoto(String nomPhoto) {
		this.nomPhoto = nomPhoto;
	}

	public List<Produit> getProduits() {
		return produits;
	}

	public void setProduits(List<Produit> produits) {
		this.produits = produits;
	}

	public Categorie() {
		super();

	}

	public Categorie(String nomCategorie, String description, String nomPhoto) {
		super();
		this.nomCategorie = nomCategorie;
		this.description = description;
		this.nomPhoto = nomPhoto;
	}

}